Schedule_app/views.py

```diff
@@ -40,22 +60,142 @@ def bookticket(response):
         form = TicketBookForm()
         context = {'form': form}
         return render(response, 'bookticket.html', context)
-    # submitted = False
-    # if response.method == "POST":
-    #     form = TicketBookForm(response.POST)
-    #     if form.is_valid():
-    #         form.save()
-    #         return HttpResponseRedirect('/book_ticket?submitted=True')
-    # else:
-    #     form = TicketBookForm
-    #     if 'submitted' in response.GET:
-    #         submitted = True
-    # return render(response, 'bookticket.html', {"form": form, 'submitted': submitted, 'logedin': True})
     else:
         return render(response, 'bookticket.html', {'logedin': False})
 
 
 def cinema(response):
     form = TicketBookForm(response.GET)
-    print(form['cinema'])
     return HttpResponse(form['cinema'])
+
+
+def movie(response):
+    form = TicketBookForm(response.GET)
+    return HttpResponse(form['movie'])
+
+
+def hall(response):
+    form = TicketBookForm(response.GET)
+    return HttpResponse(form['hall'])
+
+
+def time(response):
+    form = TicketBookForm(response.GET)
+    return HttpResponse(form['time'])
+
+
+def seats(response):
+    if response.user.is_authenticated:
+        data = response.POST
+
+        schedule = ScheduleMovieCinema.objects.filter(city=data['city'], cinema=data['cinema'], movie=data['movie'],
+                                                      hall=data['hall'], playing=data['time']).values()
+        seats_context = SeatModel.objects.filter(name__icontains='city{}'.format(data['city'])). \
+            filter(name__icontains='cinema{}'.format(data['cinema'])) \
+            .filter(name__icontains='hall{}'.format(data['hall'])). \
+            order_by('id').values()
+        reserved_seats = BookTicket.objects.filter(schedule=schedule[0]['id']).values()
+
+        print(seats_context[0])
+
+        for seat in seats_context:
+            for i in range(len(reserved_seats)):
+                if str(seat['id']) in reserved_seats[i]['seats']:
+                    seat['available'] = False
+        return render(response, 'chooseseat.html', {'seats': seats_context,
+                                                    'initial_seat': seats_context[0],
+                                                    'schedule': schedule[0]['id']})
+    else:
+        return render(response, 'chooseseat.html', {'logedin': False})
+
+
+def confirmation(response):
+    data = response.POST
+    reservations = []
+    for bookings in data:
+        reservations.append(bookings)
+    reservations.pop(0)
+    reservations.pop(len(reservations) - 1)
```

Comment:
I don't understand why you'd want to remove the first and the last reservation from the user's selection. Can you explain?

Reply:
IT was because in reservations i would also have data about csrftoken and other data that i don't need. It was for a bit of clean-up

Reply:
I think I know what you mean, but are you sure you're always going to receive them in the same order?
Normally, we should be able to access the data as `request.POST["INPUT_NAME"]`. Try to see if request.POST.keys() will help in this case.

Reply:
Checking now